File: kotlin/nth-prime/src/main/kotlin/Prime.kt
class Prime {
    companion object {
        fun nth(n: Int): Int {
            require (n > 0) {"There is no zeroth prime."}

            var candidate = 1
            var primeCounter = 1
            do {
                if (isPrime(++candidate)) {
                    if (primeCounter++ == n) {
                        return candidate
                    }
                }
            } while (true)

            return candidate
        }

        private fun isPrime(n: Int): Boolean {
            for (possibleFactor in 2..(n/2)) {
                if (n.rem(possibleFactor) == 0) {
                    return false
                }
            }
            return true
        }
    }
}

File: elixir/nth-prime/lib/prime.ex
defmodule Prime do
  @doc """
  Generates the nth prime.
  """
  @spec nth(non_neg_integer) :: non_neg_integer
  def nth(count) do
       Stream.iterate({2, true}, fn {a, _} -> {a + 1, prime?(a + 1)} end)
       |> Stream.filter(fn {_, prime_status} -> prime_status == true end)
       |> Enum.take(count)
       |> Enum.at(count - 1)
       |> elem(0)
  end

  defp prime?(number) do
    div = max(2, Integer.floor_div(number, 2))
    divisors = for n <- div..2, rem(number, n) == 0, do: n

    divisors == []
  end
end
